core/bentley/src/Config.ts

```diff
@@ -19,7 +22,7 @@
  *  i.e. If the variable `appName='TestApp'`, a variable with a value of 'https://${appName}' will be expanded to `https://TestApp`.
  * > Does not recursively expand variables such as, `appName=${appName}`.
  *
- * @public
+ * @public @deprecated

```

Comment:
I don't understand - the PR is entitled "Remove Config.App" but you haven't removed it.
Instead you have deprecated the entire Config class. Are there replacements for the deprecated functionality? Should the entire Config class be removed?

Reply:
Ah yes, my bad. The initial intention was to remove it until we realized it hadn't yet been marked as deprecated. Let me update the PR title. The 'replacement' is just accessing the variables from process.env directly which is essentially just what Config was doing.

Reply:
You're targeting master which is producing a new major version. @calebmshafer we're not allowed to directly remove something in 3.0 that wasn't marked deprecated in 2.x?

Reply:
That is the rule but we accidentally did not mark this class as intended to be deprecated. I'll leave it up to what others think is the best approach.

We would be breaking the rule here, despite that I feel it's a rather safe move since the replacement is straightforward and most, if not all, of the use-cases are in the current client libraries. Which are all going to be removed anyway...

Reply:
Well, we had planned to start work on 3.0 by deleting all the deprecated APIs; instead, we're adding new ones.
Could we release a new 2.19.x that deprecates all the stuff we want to purge, so we don't have to carry that baggage into 3.0?

Reply:
I like Paul's plan, at least for this class.

Reply:
Sounds good to me. @pbell97 mind opening a PR that marks this class as deprecated targeted at the `release/2.19.x` branch?